CMakeLists.txt

```diff
@@ -323,7 +323,7 @@ if(Python3_FOUND)
 endif()
 
 if(IREE_BUILD_TESTS AND NOT IREE_PYYAML_FOUND)
-  message(WARNING "IREE's regression test suite requires PyYAML to run all tests. It is not installed, so some tests will be disabled.")
+  message(FATAL_ERROR "IREE's regression test suite requires PyYAML. Try installing it, perhaps with a command like:  pip install pyyaml")
 endif()
```

Comment:
Why change this failure mode? pyyaml is only used for a small subset of tests

Reply:
IMO quiet failures are bad (and since no one reads warnings, that includes warnings). I think we should provide some way to disable these before making this a hard failure though.

Reply:
I read warnings, and this one bugs me enough already - I'd rather it not change to a hard failure.

How about disabling / skipping the relevant tests so they still show up in ctest's output as skipped? Some ideas for implementations in here:
* https://stackoverflow.com/questions/61572034/ctest-disable-a-set-of-labeled-tests-by-default-but-run-them-when-explicitly
* https://stackoverflow.com/questions/54525943/skipping-a-ctest-if-the-target-was-not-built